### DISCUSSION TITLE
Cmnd+Click to see definition and Rename symbol feature

### DISCUSSION THREAD

**Discussion by faroukcharkas** [astral-sh/ruff#12460] [Ideas]
Pylance is slow and crashes all the time, so I wanted to find something much faster and more powerful than Pylance. I found Ruff, which has been incredible for identifying and quickly fixing issues. I love it so much and I recommend it to everyone. I want to completely leave Pylance in totality, but their VSCode feature where you can use `Cmd+Click` and see the function definition, or the `F2` feature where you can rename a smybol is ultimately incredibly necessary to my developer workflow and would love ditch Pylance and use only Ruff.

I'm not sure if this has already been made but from my research it hasn't. This isn't pressing at all, but would greatly increase my (and probably many others) quality of life!

Love Ruff!

**Comment by dhruvmanila** [comment]
Thank you for the kind words!

We're currently in the middle of completely changing the architecture of Ruff to support multifile analysis which will unlock different capabilities in the language server which powers the VS Code extension. This is going to take some time as it's a lot of work but we do want to expand our language server with additional capabilities like goto definition, rename, etc.
